src/server/models/order.ts

```diff
@@ -32,30 +32,105 @@ class Order extends Model<Order> {
+        //   }
+        // }
         return cart
       })
-      .then(cart => OrdersProducts.create({ orderId: cart.id, productId: productId }))
+      // 4. find existing ordersProduct instance and increase quantity by quantity.
+      // 5. if no existing ordersProduct instance, create instance.
+      .then(cart => {
+        // cart = order with status cart
+        return OrdersProducts.findAll({
+          where: {
+            orderId: cart.id
+          }
+        }).then(async (userOrderCart) => {
+          // userOrderCart is the orderProducts for the user's cart.
+          // If the productId exists, add quantity, else create product with that qty.
+          let foundProductFromCart = userOrderCart.filter(orderProdLineItem => orderProdLineItem.productId === productId)
+          if (foundProductFromCart.length === 0) {
+            // product does not exist in cart. add product to order
+            return await OrdersProducts.create({ orderId: cart.id, productId: productId, quantity: quantity })
+          } else {
+            // product exists in cart. increase quantity
+            let rowToUpdate = foundProductFromCart[0]
+            return await rowToUpdate.update({ quantity: rowToUpdate.quantity + quantity })
+          }
+        })
+      })
       .catch((e: Error) => console.log(`Failed to add to cart. \n${e}`))
   }
+
+  // Removes a product from cart. if optional paramter removeAll=true, then it removes all quantity of this item from the cart
+  static decreaseProductQuantityFromCart(userId: number, productId: number, removeAll: boolean = false) {
+    // find user's cart
+    return Order.findOne({
+      where: {
+        userId: userId,
+        status: Status.Cart
+      }
+    })
+      .then(cart => {
+        if (!cart) throw new Error(`User ${userId} has no cart. Should not empty a nonexistant cart.`)
+        OrdersProducts.findOne({
+          where: {
+            orderId: cart.id,
+            productId: productId
+          }
+        })
+          .then(async (orderProductLineItem) => {
+            if (!orderProductLineItem) throw new Error(`Product ${productId} does not exist in cart.`)
+            if (orderProductLineItem.quantity === 1 || removeAll) return await orderProductLineItem.destroy()
+            return await orderProductLineItem.update({ quantity: orderProductLineItem.quantity - 1 })
+          })
+          .catch(er => console.log(`Failed to ${removeAll ? 'delete' : 'decrement'} product ${productId} from ${userId}'s cart.`))
+      })
+  }
+
+  static emptyCart(userId: number) {
+    // removes all items from a user's cart.
+    // finds a user's cart. removes all associated orderProduct 
+    return Order.findOne({
+      where: {
+        userId: userId,
+        status: Status.Cart
+      }
+    })
+      .then(cart => {
+        if (!cart) throw new Error(`User ${userId} has no cart. Should not empty a nonexistant cart`)
+        OrdersProducts.findAll({
```

Comment:
And again here: we need to return this Promise. Also, try looking at this syntax from the Sequelize docs: 
`Task.bulkCreate([
  {subject: 'programming', status: 'executing'},
  {subject: 'reading', status: 'executing'},
  {subject: 'programming', status: 'finished'}
])`
`.then(() => {
  return (We should use this!) Task.destroy({
    where: {
      subject: 'programming'
    },
    truncate: true /* this will ignore where and truncate the table instead */
  });
})`
`.then(affectedRows => {
  // affectedRows will be 2
  return Task.findAll();
})`
`.then(tasks => {
  console.log(tasks) // no programming, just reading :(
})`